index.js

```diff
@@ -80,10 +80,26 @@ function resolver(src) {
 
         loader = typeof element.addEventListener !== 'undefined' ? loadStyle : /* istanbul ignore next */ loadStyleIE;
         resolve(loader(element));
+        head.appendChild(element);
+      } else if (/\.json$/.test(src.url)) {
+        var xhr = new XMLHttpRequest();
+        xhr.open('GET', src.url, true);
+        xhr.onreadystatechange = function () {
+          if (xhr.readyState === XMLHttpRequest.DONE) {
+            if (xhr.status === 200) {
+              resolve(JSON.parse(xhr.responseText));
+            } else if (xhr.status >= 400) {
+              reject(new Error('Failed to load ' + src.url));
+            }
+          }
+        }
+        xhr.withCredentials = !!src.withCredentials;
+        xhr.send();
       } else {
         element = doc.createElement('script');
         element.charset = 'utf8';
         element.src = src.url;
+        element.async = true;
```

Comment:
Make this an option so we can guarantee there won't be any impact from this change.